ctapipe/tools/make_irf.py

```diff
@@ -154,11 +154,25 @@ def make_derived_columns(self, kind, events, spectrum, target_spectrum, obs_conf
     def get_metadata(self, loader):
         obs = loader.read_observation_information()
         sim = loader.read_simulation_configuration()
+        show = loader.read_shower_distribution()
 
         # These sims better have the same viewcone!
+        if not np.diff(sim["energy_range_max"]).sum() == 0:
```

Comment:
To make intent clearer here, this could be `len(np.unique(...)) > 0` for each of these.

You can also test them in a loop over the keys.

Reply:
implemented.